search-provider/eks-discovery-feed-provider.c

```diff
@@ -223,19 +223,15 @@ get_day_of_year (void)
   return g_date_time_get_day_of_year (datetime);
 }
 
-static gchar *
-select_string_from_variant_from_day (GVariant *variant)
+static const char *
+select_string_from_array_from_day (JsonArray *array)
 {
-  gsize size = g_variant_n_children (variant);
+  guint size = json_array_get_length (array);
   if (size == 0)
     return NULL;
 
   gint index = get_day_of_week () % size;
```

Comment:
Not part of the patch, but I'd strongly recommend using `idx` or `index_` instead of `index`, in this code. Yes, there's an `index()` function in the C standard library, declared in `string.h`. You should be seeing compiler warnings when building.

Reply:
I :heart: my `ix`